client/engine/engine.go

```diff
@@ -122,6 +125,10 @@ func (e *Engine) ToApi() <-chan EngineEvent {
 	return e.toApi
 }
 
+func (e *Engine) Stop() {
```

Comment:
What do you think about `Close` instead of `Stop`? I think `Close` may be more [idiomatic](https://pkg.go.dev/io#Closer) for golang.